<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>qpatternfix</class>
 <widget class="QFrame" name="qpatternfix">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>688</width>
    <height>476</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Frame</string>
  </property>
  <widget class="QWidget" name="horizontalLayoutWidget">
   <property name="geometry">
    <rect>
     <x>12</x>
     <y>8</y>
     <width>631</width>
     <height>38</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QLabel" name="label_fix_main">
      <property name="font">
       <font>
        <pointsize>11</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Pattern-Fix Editor</string>
      </property>
     </widget>
    </item>
    <item>
     <spacer name="horizontalSpacer">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <widget class="QLabel" name="label_pattern_no">
      <property name="minimumSize">
       <size>
        <width>100</width>
        <height>0</height>
       </size>
      </property>
      <property name="font">
       <font>
        <italic>true</italic>
       </font>
      </property>
      <property name="text">
       <string>Pattern No. </string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="label_pattern">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="minimumSize">
       <size>
        <width>56</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>56</width>
        <height>28</height>
       </size>
      </property>
      <property name="font">
       <font>
        <italic>true</italic>
       </font>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Sunken</enum>
      </property>
      <property name="lineWidth">
       <number>2</number>
      </property>
      <property name="midLineWidth">
       <number>0</number>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="label">
      <property name="minimumSize">
       <size>
        <width>108</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>64</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="font">
       <font>
        <italic>true</italic>
       </font>
      </property>
      <property name="text">
       <string>Max. Tick</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="label_pulses">
      <property name="minimumSize">
       <size>
        <width>96</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>64</width>
        <height>28</height>
       </size>
      </property>
      <property name="font">
       <font>
        <italic>true</italic>
       </font>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Sunken</enum>
      </property>
      <property name="lineWidth">
       <number>2</number>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QGroupBox" name="group_box_length">
   <property name="geometry">
    <rect>
     <x>8</x>
     <y>56</y>
     <width>261</width>
     <height>151</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <bold>true</bold>
    </font>
   </property>
   <property name="title">
    <string>Length Change</string>
   </property>
   <property name="flat">
    <bool>false</bool>
   </property>
   <widget class="QWidget" name="layoutWidgetLength">
    <property name="geometry">
     <rect>
      <x>9</x>
      <y>33</y>
      <width>259</width>
      <height>104</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <property name="horizontalSpacing">
      <number>22</number>
     </property>
     <item row="1" column="0">
      <widget class="QRadioButton" name="btn_change_pick">
       <property name="font">
        <font>
         <bold>false</bold>
        </font>
       </property>
       <property name="toolTip">
        <string>If an integer, the number of measures is
changed; events are compressed or expanded
as needed.

If a fraction (e.g. 3/4), the pattern is reduced to
one measure with the given time signature.</string>
       </property>
       <property name="text">
        <string>&amp;Measures/Time Sig</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QRadioButton" name="btn_change_scale">
       <property name="font">
        <font>
         <bold>false</bold>
        </font>
       </property>
       <property name="toolTip">
        <string>Shrink/expand the pattern by a factor,
accounting for the time of the last event.</string>
       </property>
       <property name="text">
        <string>Scale &amp;Factor</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QRadioButton" name="btn_change_none">
       <property name="font">
        <font>
         <bold>false</bold>
        </font>
       </property>
       <property name="toolTip">
        <string>No change in pattern length.</string>
       </property>
       <property name="text">
        <string>&amp;None</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLineEdit" name="line_edit_measures">
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="line_edit_none">
       <property name="enabled">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLineEdit" name="line_edit_scale"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QGroupBox" name="group_box_effect">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>530</x>
     <y>56</y>
     <width>141</width>
     <height>221</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <italic>true</italic>
     <bold>true</bold>
    </font>
   </property>
   <property name="title">
    <string>Applied Effects</string>
   </property>
   <property name="flat">
    <bool>false</bool>
   </property>
   <property name="checkable">
    <bool>false</bool>
   </property>
   <widget class="QWidget" name="layoutWidgetEffect">
    <property name="geometry">
     <rect>
      <x>8</x>
      <y>35</y>
      <width>121</width>
      <height>181</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QCheckBox" name="btn_effect_alteration">
       <property name="font">
        <font>
         <italic>true</italic>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>Alteration</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="btn_effect_shift">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="font">
        <font>
         <italic>true</italic>
         <bold>false</bold>
        </font>
       </property>
       <property name="toolTip">
        <string>Events will be shifted.</string>
       </property>
       <property name="text">
        <string>Shift</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="btn_effect_reverse">
       <property name="font">
        <font>
         <italic>true</italic>
         <bold>false</bold>
        </font>
       </property>
       <property name="toolTip">
        <string>Events will be reversed.</string>
       </property>
       <property name="text">
        <string>Reverse</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="btn_effect_shrink">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="font">
        <font>
         <italic>true</italic>
         <bold>false</bold>
        </font>
       </property>
       <property name="toolTip">
        <string>Events will be compressed.</string>
       </property>
       <property name="text">
        <string>Shrin&amp;k</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="btn_effect_expand">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="font">
        <font>
         <italic>true</italic>
         <bold>false</bold>
        </font>
       </property>
       <property name="toolTip">
        <string>Events will expand.</string>
       </property>
       <property name="text">
        <string>E&amp;xpand</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="btn_effect_time_sig">
       <property name="font">
        <font>
         <italic>true</italic>
         <bold>false</bold>
        </font>
       </property>
       <property name="toolTip">
        <string>The time signature will change.</string>
       </property>
       <property name="text">
        <string>Time Sig</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QGroupBox" name="group_box_other">
   <property name="geometry">
    <rect>
     <x>8</x>
     <y>220</y>
     <width>221</width>
     <height>201</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <bold>true</bold>
    </font>
   </property>
   <property name="title">
    <string>Other Fixes</string>
   </property>
   <widget class="QWidget" name="layoutWidgetOtherFixes">
    <property name="geometry">
     <rect>
      <x>12</x>
      <y>24</y>
      <width>201</width>
      <height>161</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="QCheckBox" name="btn_align_left">
       <property name="font">
        <font>
         <bold>false</bold>
        </font>
       </property>
       <property name="toolTip">
        <string>Move events left to remove delays.</string>
       </property>
       <property name="text">
        <string>&amp;Align left</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="btn_align_right">
       <property name="font">
        <font>
         <bold>false</bold>
        </font>
       </property>
       <property name="text">
        <string>Align right</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="btn_reverse">
       <property name="font">
        <font>
         <bold>false</bold>
        </font>
       </property>
       <property name="toolTip">
        <string>Reverse the order of all events, flipping
the measures.</string>
       </property>
       <property name="text">
        <string>Reverse measures</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="btn_reverse_in_place">
       <property name="font">
        <font>
         <bold>false</bold>
        </font>
       </property>
       <property name="toolTip">
        <string>Reverse notes, preserving relative location.</string>
       </property>
       <property name="text">
        <string>Reverse in place</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="btn_save_note_length">
       <property name="font">
        <font>
         <bold>false</bold>
        </font>
       </property>
       <property name="toolTip">
        <string>When scaling, don't change note length.</string>
       </property>
       <property name="text">
        <string>&amp;Preserve note length</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QWidget" name="layoutWidgetAlignLeft">
   <property name="geometry">
    <rect>
     <x>250</x>
     <y>380</y>
     <width>381</width>
     <height>38</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <widget class="QPushButton" name="btn_set">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>28</height>
       </size>
      </property>
      <property name="toolTip">
       <string>Apply the settings.</string>
      </property>
      <property name="text">
       <string>&amp;Set</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="btn_reset">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>28</height>
       </size>
      </property>
      <property name="toolTip">
       <string>Go back to the original pattern.</string>
      </property>
      <property name="text">
       <string>&amp;Reset</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="btn_close">
      <property name="minimumSize">
       <size>
        <width>28</width>
        <height>0</height>
       </size>
      </property>
      <property name="toolTip">
       <string>Close this dialog.</string>
      </property>
      <property name="text">
       <string>&amp;Close</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QGroupBox" name="group_box_quantize">
   <property name="geometry">
    <rect>
     <x>270</x>
     <y>56</y>
     <width>258</width>
     <height>321</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="font">
    <font>
     <bold>true</bold>
    </font>
   </property>
   <property name="title">
    <string>Alteration</string>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_3">
    <item>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="7" column="0">
       <widget class="QRadioButton" name="btn_alt_rev_notemap">
        <property name="font">
         <font>
          <bold>false</bold>
         </font>
        </property>
        <property name="toolTip">
         <string>Remap notes in the opposite direction of &quot;Note-map&quot;.</string>
        </property>
        <property name="text">
         <string>Rev Note-map</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QRadioButton" name="btn_alt_none">
        <property name="font">
         <font>
          <bold>false</bold>
         </font>
        </property>
        <property name="toolTip">
         <string>No alteration operations will be done.</string>
        </property>
        <property name="text">
         <string>None</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="line_edit_alt_tighten">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="maximumSize">
         <size>
          <width>48</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="toolTip">
         <string>If not blank, replaces the snap/2 value.</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QRadioButton" name="btn_alt_tighten">
        <property name="font">
         <font>
          <bold>false</bold>
         </font>
        </property>
        <property name="toolTip">
         <string>Do partial quantization.</string>
        </property>
        <property name="text">
         <string>&amp;Tight Q ticks</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QRadioButton" name="btn_alt_full">
        <property name="font">
         <font>
          <bold>false</bold>
         </font>
        </property>
        <property name="toolTip">
         <string>Fully quantization to the snap value.</string>
        </property>
        <property name="text">
         <string>F&amp;ull Q ticks</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QPushButton" name="btn_notemap_file">
        <property name="maximumSize">
         <size>
          <width>48</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>...</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QRadioButton" name="btn_alt_jitter">
        <property name="font">
         <font>
          <bold>false</bold>
         </font>
        </property>
        <property name="toolTip">
         <string>Jitter the timestamps of the events.</string>
        </property>
        <property name="text">
         <string>Time/tick jitter</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QRadioButton" name="btn_alt_notemap">
        <property name="font">
         <font>
          <bold>false</bold>
         </font>
        </property>
        <property name="toolTip">
         <string>Remap notes as per the loaded 'drum' file.</string>
        </property>
        <property name="text">
         <string>Note-map</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QRadioButton" name="btn_alt_random">
        <property name="font">
         <font>
          <bold>false</bold>
         </font>
        </property>
        <property name="toolTip">
         <string>Randomize event magnitude or note velocity.</string>
        </property>
        <property name="text">
         <string>Random amplitude</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="line_edit_alt_none">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="maximumSize">
         <size>
          <width>48</width>
          <height>16777215</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QLineEdit" name="line_edit_alt_jitter">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="maximumSize">
         <size>
          <width>48</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Plus/minus range of time jitter in ticks.</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="line_edit_alt_full">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="maximumSize">
         <size>
          <width>48</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="toolTip">
         <string>If not blank, replaces the snap value.</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLineEdit" name="line_edit_alt_random">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="maximumSize">
         <size>
          <width>48</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Plus/minus range of magnitude change in MIDI units.</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QRadioButton" name="btn_alt_random_pitch">
        <property name="font">
         <font>
          <bold>false</bold>
         </font>
        </property>
        <property name="text">
         <string>Random pitch</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QLineEdit" name="line_edit_alt_random_pitch">
        <property name="maximumSize">
         <size>
          <width>48</width>
          <height>16777215</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <tabstops>
  <tabstop>btn_change_none</tabstop>
  <tabstop>line_edit_none</tabstop>
  <tabstop>btn_change_pick</tabstop>
  <tabstop>line_edit_measures</tabstop>
  <tabstop>btn_change_scale</tabstop>
  <tabstop>line_edit_scale</tabstop>
  <tabstop>btn_align_left</tabstop>
  <tabstop>btn_set</tabstop>
  <tabstop>btn_reset</tabstop>
  <tabstop>btn_close</tabstop>
  <tabstop>btn_effect_expand</tabstop>
  <tabstop>btn_effect_shrink</tabstop>
  <tabstop>btn_effect_shift</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
